src/language-js/loc.js

```diff
@@ -4,20 +4,16 @@ import isNonEmptyArray from "../utils/is-non-empty-array.js";
  * @typedef {import("./types/estree.js").Node} Node
  */
 
-function locStart(node, opts) {
-  const { ignoreDecorators } = opts || {};
+function locStart(node) {
+  const start = node.range ? node.range[0] : node.start;
 
   // Handle nodes with decorators. They should start at the first decorator
-  if (!ignoreDecorators) {
-    const decorators =
-      (node.declaration && node.declaration.decorators) || node.decorators;
-
-    if (isNonEmptyArray(decorators)) {
-      return locStart(decorators[0]);
-    }
+  const decorators = node.declaration?.decorators ?? node.decorators;
+  if (isNonEmptyArray(decorators)) {
+    return Math.min(locStart(decorators[0]), start);
```

Comment:
It seems the old one is not right, if decorators after `export`, it will still use range of `decorators[0]`

Reply:
In which case is `locStart(decorators[0])` less then `start`?

Reply:
> In which case is locStart(decorators[0]) less then start?

Other parsers, `flow`/`typescript` I think they followed old behavior of babel.